services/web/client/source/resource/qxapp/node-meta-v0.0.1.json

```diff
@@ -0,0 +1,327 @@
+        },
+        "type": {
+          "type": "string",
+          "description": "service type",
+          "enum": ["computational","dynamic"],
+          "examples": [
+              "computational"
+          ]
+        },
+        "name": {
+            "type": "string",
+            "description": "short, human readable name for the node",
+            "examples": [
+                "Fast Counter"
+            ]
+        },
+        "description": {
+            "type": "string",
+            "description": "human readable description of the purpose of the node",
+            "examples": [
+                "Our best node type",
+                "The mother of all nodes, makes your numbers shine!"
+            ]
+        },
+        "authors": {
+            "type": "array",
+            "minItems": 1,
+            "items": {
+                "type": "object",
+                "required": ["name", "email"],
+                "additionalProperties": false,
+                "properties": {
+                    "name": {
+                        "type": "string",
+                        "description": "Name of the author",
+                        "examples": [
+                            "Sun Bak",
+                            "Delenn"
+                        ]
+                    },
+                    "email": {
+                        "description": "Email address",
+                        "type": "string",
+                        "format": "email",
+                        "examples": [
+                            "sun@sense.eight",
+                            "deleen@minbar.bab"
+                        ]
+                    },
+                    "affiliation": {
+                        "description": "Affiliation of the author",
+                        "type": "string",
+                        "examples": [
+                            "Sense8",
+                            "Babylon 5"
+                        ]
+                    }
+                }
+            }
+        },
+        "contact": {
+            "type": "string",
+            "format": "email",
+            "description": "email to correspond to the authors about the node",
+            "examples": [
+                "lab@net.flix"
+            ]
+        },
+        "inputs": {
+            "type": "object",
+            "description": "definition of the inputs of this node",
+            "patternProperties": {
+                "^[-_a-zA-Z0-9]+$": {
+                    "type": "object",
+                    "description": "all the input configurable for this service",
+                    "additionalProperties": false,
+                    "required": [
+                        "displayOrder",
+                        "label",
+                        "description",
+                        "type"
+                    ],
+                    "properties": {
+                        "displayOrder": {
+                            "type": "number",
```

Comment:
Is this really necessary, can't we just define by order of appearance?

Reply:
having keys to access inputs and outputs is much simpler than accessing them by position. The key could be a property of the input/output, but then the data format itself could not guard against duplicate key mentions which it does now by virtue of the key being ... a key.

otoh, a hash is not sorted, so we need the ordering property for sorting ... 

both approaches have their weaknesses ... I judged the integrity of the keyset to be higher than the added effort of sorting ...